addons/website_event_exhibitor/controllers/exhibitor.py

```diff
@@ -199,9 +199,11 @@ def event_sponsor_read(self, sponsor_id):
         else:
             sponsor_data['country_name'] = False
             sponsor_data['country_id'] = False
-        if sponsor.sponsor_type_id:
-            sponsor_data['sponsor_type_name'] = sponsor.sponsor_type_id.name
-            sponsor_data['sponsor_type_id'] = sponsor.sponsor_type_id.id
+        # needs sudo access as public users can't read the model
+        sponsor_type = sponsor.sponsor_type_id.sudo()
+        if sponsor_type:
```

Comment:
Seems we don't need the if (False being the value given by a void record) ?

Reply:
Indeed, also after double checking the field is actually configured as required, so there was really no reason for this check.

Simplified!